File: researchflow/backend/app/schemas/research.py
from pydantic import BaseModel
from typing import Optional, List
from datetime import datetime


class ResearchRequest(BaseModel):
    topic: str
    depth: str = "medium"


class ResearchResponse(BaseModel):
    session_id: str
    status: str
    message: str


class StatusResponse(BaseModel):
    session_id: str
    status: str
    progress: int
    current_agent: Optional[str] = None
    report_id: Optional[str] = None
    error_message: Optional[str] = None


class SourceDict(BaseModel):
    url: str
    title: str
    snippet: str


class ReportResponse(BaseModel):
    report_id: str
    session_id: str
    topic: str
    content: str
    sources: List[SourceDict]
    word_count: int
    created_at: datetime

File: researchflow/frontend/src/api/research.js
import axios from 'axios';
import { API_BASE_URL } from '../utils/constants';

const api = axios.create({
  baseURL: API_BASE_URL,
  headers: { 'Content-Type': 'application/json' },
});

/**
 * Start a new research session (legacy standalone endpoint).
 * @param {string} topic
 * @param {string} depth - 'shallow' | 'medium' | 'deep'
 * @returns {Promise<{ session_id: string, status: string, message: string }>}
 */
export const createResearch = async (topic, depth = 'medium') => {
  const response = await api.post('/api/research', { topic, depth });
  return response.data;
};

/**
 * Poll the status of an ongoing or finished research session.
 * @param {string} sessionId
 * @returns {Promise<{ session_id, status, progress, current_agent, report_id, error_message }>}
 */
export const getResearchStatus = async (sessionId) => {
  const response = await api.get(`/api/status/${sessionId}`);
  return response.data;
};

/**
 * Fetch a completed report by its ID.
 * @param {string} reportId
 * @returns {Promise<{ report_id, session_id, topic, content, sources, word_count, created_at }>}
 */
export const getReport = async (reportId) => {
  const response = await api.get(`/api/report/${reportId}`);
  return response.data;
};
